cowbird/services/impl/geoserver.py

```diff
@@ -38,156 +81,174 @@ def __init__(self, settings, name, **kwargs):
         """
         super(Geoserver, self).__init__(settings, name, **kwargs)
         self.api_url = "{}/rest".format(self.url)
-        self.auth = (self.admin_user, self.admin_password)
         self.headers = {"Content-type": CONTENT_TYPE_JSON}
+        self.admin_user = kwargs.get(SERVICE_ADMIN_USER, None)
+        self.admin_password = kwargs.get(SERVICE_ADMIN_PASSWORD, None)
+        self.auth = (self.admin_user, self.admin_password)
+
+    #
+    # Implementation of parent classes' functions
+    #
 
+    # Service class functions
     def get_resource_id(self, resource_full_name):
-        # type (str) -> str
+        # type:(str) -> str
         raise NotImplementedError
 
     def user_created(self, user_name):
-        res = chain(create_workspace.si(user_name) | create_datastore.si(user_name))
+        self._create_datastore_dir(user_name)
+        res = chain(create_workspace.si(user_name), create_datastore.si(user_name))
         res.delay()
+        LOGGER.info("Start monitoring datastore of created user [%s]", user_name)
+        Monitoring().register(self._shapefile_folder_dir(user_name), True, Geoserver)
 
     def user_deleted(self, user_name):
         remove_workspace.delay(user_name)
+        LOGGER.info("Stop monitoring datastore of created user [%s]", user_name)
+        Monitoring().unregister(self._shapefile_folder_dir(user_name), self)
 
     def permission_created(self, permission):
         raise NotImplementedError
 
     def permission_deleted(self, permission):
         raise NotImplementedError
 
+    # FSMonitor class functions
+    @staticmethod
+    def get_instance():
```

Comment:
Why creating this function if you still use `ServiceFactory().get_service("Geoserver")` everywhere 😜

Reply:
Ooops haha! The Tasks where created before I implemented the dual inheritance with FSMonitor class and I must have implemented the abstract function mechanically from the Catalog implementation without thinking about it.

Looking at it now however, that function should return a monitor instance, and in both `geoserver.py` and `catalog.py`, we use it to return a Service instance. Is this really the behavior expected? If so I'd work on the description a bit to make it clearer.

Reply:
Sorry, I didn't realize it's a FSMonitor required function. The implementation is right though, that function is used here : https://github.com/Ouranosinc/cowbird/blob/master/cowbird/monitoring/monitor.py#L62 and should return an instance of `FSMonitor` which is really, in this case, a `Geoserver` instance.

Reply:
The pattern of `get_fsmonitor_instance` -> `get_instance` and similar feature for GeoServer is telling me `get_instance` should probably be an abstract method that classes must override with their own implementations.

Reply:
Oh, I mixed `FSMonitor` and `Monitor` classes. All good.